pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/PublishDraftVersion.java

```diff
@@ -60,16 +63,16 @@ public PublishDraftVersion(@Reference Events events,
         this.serviceResourceResolverProvider = serviceResourceResolverProvider;
     }
 
-    private Module getModule(SlingHttpServletRequest request) {
-        return request.getResource().adaptTo(Module.class);
+    private Document getDocument(SlingHttpServletRequest request) {
+        return request.getResource().adaptTo(Document.class);
     }
 
     private Locale getLocale(SlingHttpServletRequest request) {
         return paramValueAsLocale(request, "locale", GlobalConfig.DEFAULT_MODULE_LOCALE);
     }
 
     private String getVariant(SlingHttpServletRequest request) {
-        return paramValue(request, "variant", ModuleVariant.DEFAULT_VARIANT_NAME);
+        return paramValue(request, "variant", DocumentVariant.DEFAULT_VARIANT_NAME);
     }
```

Comment:
Not your code, but I worry that we have a bug here, so also tagging @carlosmunoz @xdavidson @aprajshekhar.

I don't think it's appropriate to assume that if the caller doesn't specify a variant parameter for publishing, that we assume DEFAULT_VARIANT_NAME.

Better options:
- Assume `document.getWorkspace().getCanonicalVariantName()`
- Throw an exception, because if the caller is publishing something, they should be explicit about which variant they want rather than us potentially guessing wrong.

Thoughts / other opinions?

Reply:
For the record, the same method appears in UnpublishVersion.java as well.

Reply:
@benradey since this logic appears at multiple files, I would say let's table it till MS3 is done. Creating a JIRA to track this is what I would suggest and lets discuss the options there. Discussing it here would need updating the JIRA and that would be a overhead. Just my two cents.